src/CodeGen_Hexagon.cpp

```diff
@@ -1716,7 +1716,6 @@ void CodeGen_Hexagon::visit(const Call *op) {
             internal_assert(op->args.size() == 2);
             string instr = op->is_intrinsic(Call::shift_left) ? "halide.hexagon.shl" : "halide.hexagon.shr";
             Expr b = maybe_scalar(op->args[1]);
-            internal_assert(b.type().is_uint());
```

Comment:
I think this assert should remain, until we know that Hexagon's shift operators do the right thing with negative shifts. Until then, this should use `lower_signed_shift_*` to avoid shifts by signed RHS values.

I think you could implement this by inserting this before the assert:

    if (b.type().is_int()) {
        CodeGen_Posix::visit(op);
        return;
    }